contracts/src/v0.8/shared/test/call/CallWithExactGas.t.sol

```diff
@@ -5,19 +5,26 @@ import {CallWithExactGas} from "../../call/CallWithExactGas.sol";
 import {CallWithExactGasHelper} from "./CallWithExactGasHelper.sol";
 import {BaseTest} from "../BaseTest.t.sol";
 import {GenericReceiver} from "../testhelpers/GenericReceiver.sol";
+import {GasConsumer} from "../testhelpers/GasConsumer.sol";
 
 contract CallWithExactGasSetup is BaseTest {
   GenericReceiver internal s_receiver;
   CallWithExactGasHelper internal s_caller;
+  GasConsumer internal s_gasConsumer;
+
   uint256 internal constant DEFAULT_GAS_LIMIT = 20_000;
   uint16 internal constant DEFAULT_GAS_FOR_CALL_EXACT_CHECK = 5000;
   uint256 internal constant EXTCODESIZE_GAS_COST = 2600;
 
+  // Calculated as the difference between gasUsed as returned from function and gas passed in to the function
+  uint256 internal constant CALL_WITH_EXACT_GAS_SAFE_RETURN_DATA_GAS_OVERHEAD = 128;
```

Comment:
can you explain why the `difference between gasUsed as returned from function and gas passed in to the function` turns out to be 128, btw is this overhead constant regardless of receiver logic?

Reply:
I'm not able to say that with complete certainty due to optimizations in the compiler that may insert additional unexpected bytecode. The only way to get that level of absolute certainty would be to write a minimal huff-contract to ensure that every opcode is expected. `128 gas` was determined as overhead because that was the difference in value returned from SafeCallWithExactGas. That function was called with 20k gas, but the `gasUsed` returned was `20,128`. Since its impossible for the contract to use more than the gas it was assigned, the only logical explanation was that there's additional gas overhead in the call that isn't being properly accounted for.

Reply:
I see, looking at the code, it's likely constant, can you:
1. pseudo-verify it is constant regardless of gas limit/payload/receiver by trying out a few different inputs
2. succinctly explain in comments the heuristics used to arrive at 18, similar to context above

Reply:
I'll do my best. Without inspecting the bytecode manually or writing it in Huff it's very difficult to know exactly how much it takes cause of the compiler optimizations. I will write out the comments and see what I can do.